src/libstore/globals.cc

```diff
@@ -125,6 +125,57 @@ Settings::Settings()
     };
 }
 
+// FIXME: this is a duplicate from eval-settings.hh, but we can't import that here
+// if we want to maintain proper separation of concerns
```

Comment:
"Separation of concerns" is less important than avoiding code duplication. If this function is used/defined in `libstore`, then `libexpr` may as well use it.

However, it seems really hacky to move NIX_PATH handling into `libstore`, especially since we've been trying to move settings *out* of `globals.cc`. Is there really no way to avoid that?

Reply:
I think we could put another virtual function in `Setting` and call those on all the settings to load environment-based values after the config files.
That moves the environment handling to the `nixPath` setting.

I'd be ok to merge this fix first so that we don't have to rush this part of the settings design. Also we're releasing this week, so it'd be good to get this in without much delay.

Reply:
Yeah, re-doing the `Settings` as @roberth proposes it is the way to go for the mid term, but right now that would just make things later and introduce more back and forth.

For now I'd also prefer to get the fix and the tests in, and then clean up later.